### ISSUE TITLE
File size filtering causing long delay in loading sources

### ISSUE THREAD

**Issue by dv8edroute** [host505/plugin.video.theoath#16]
When file size filter is turned on it causes the sources to load extremely slowly. When disabled sources load very quickly. It is repeatable with shows or movies.

**Comment by dv8edroute** [comment]
Seems like issue is in sources.py under modules. lines 846 - 850. The i.get('size') is sometimes returning a NoneType value from the dict and causing it to hang.

I added a code snippet to change the NoneType to a float of 0.0 to fix the issue. I am a novice coder but my testing shows it solves the issue. code below

`        if size_filters == 'true':
            if _content == 'movie':
                
                for i in _sources:
                    if i.get('size') is None:
                        i['size'] = 0.0
                
                _sources = [i for i in _sources if (mov_min_size <= i.get('size') <= mov_max_size) or i.get('size', 0.0) == 0.0]

                    
            elif _content == 'episode':
                for i in _sources:
                    if i.get('size') is None:
                        i['size'] = 0.0

                _sources = [i for i in _sources if (ep_min_size <= i.get('size') <= ep_max_size) or i.get('size', 0.0) == 0.0]`

**Comment by host505** [comment]
Hi, thanks for reporting, can't say I ever noticed any difference though, albeit your reasoning does make sense.
Does simply setting the second .get() argument for size also fixes your issue?
Ie i.get('size', 0.0)

**Comment by host505** [comment]
Actually setting the 2nd value of .get() as 0.0 in the comparison seems to work. I also tried your solution but it messes with counting on the progress dialog.
Pls test the new version.

**Comment by dv8edroute** [comment]
Thanks for you rapid reply to this issue. I can confirm that using the sources.py from 3.6.001 does fix the issue on Windows and Android. A side note thanks for showing me a better way to fix an issue like that. Lastly keep up the amazing work!!!

Dv8edRoute

**Comment by host505** [comment]
For the record, the problem wasn't links reporting NoneType as value for size. These are handled in scraper level and fallback to 0.0 if size parsing fails.
The problem was scrapers that don't even have 'size' key. That's where the fallback value of .get()'s 2nd arg helped.
Ps I remember having it like this when I first introduced the size filtering function, now I have to remember when I removed it and why.
Also it didn't cause this problem on kodi 18.x/py2.
Thanks for reporting back, closing now.
